MonoGame.Framework/DrawableGameComponent.cs

```diff
@@ -25,8 +25,7 @@ public int DrawOrder
                 if (_drawOrder != value)
                 {
                     _drawOrder = value;
-                    if (DrawOrderChanged != null)
-                        DrawOrderChanged(this, null);
+                    EventHelper.Raise(this, DrawOrderChanged, null);
                     OnDrawOrderChanged(this, null);
```

Comment:
Here, instead of `EventHelper.Raise(this, DrawOrderChanged, null);` we should raise the event from within `OnDrawOrderChanged(this, null);`.

The dev should be able to override `OnDrawOrderChanged()` instead of listening to `DrawOrderChanged` and also discard the event from rising  if he chooses so (by not calling `base.OnDrawOrderChanged()`). 

https://msdn.microsoft.com/en-us/library/microsoft.xna.framework.drawablegamecomponent.ondraworderchanged.aspx

Same goes for `OnVisibleChanged()`